src/inference/Inference.jl

```diff
@@ -593,4 +594,105 @@ end
 DynamicPPL.getspace(spl::Sampler) = getspace(spl.alg)
 DynamicPPL.inspace(vn::VarName, spl::Sampler) = inspace(vn, getspace(spl.alg))
 
+"""
+    predict(model::Turing.Model, chain::MCMCChains.Chains; include_all=false)
+
+Predicts variables present in `model` but not present in `chain`, and returns a `Chains` with:
+- the predicted variables only (if `include_all == false`)
+- all variables in `model`, including variables fixed from `chain` (if `include_all == true`)
+
+In a bit more detail, the process is as follows:
+1. For every `sample` in `chain`
+   1. For every variable in `sample`
+      1. Set `variable` in `model` to its value in `sample`
+   2. Execute `model` with variables fixed as above, sampling variables NOT present in `chain` using `SampleFromPrior`
+
+#### Example
+```julia
+@model function linear_reg(x, y, σ = 0.1)
+    β ~ Normal(0, 1)
+
+    for i ∈ eachindex(y)
+        y[i] ~ Normal(β * x[i], σ)
+    end
+end
+
+xs = 0:0.1:10; ys = 2 .* xs .+ 0.1 .* randn(length(xs));
+
+# Infer
+m_lin_reg = linear_reg(xs, ys);
+chain_lin_reg = sample(m_lin_reg, NUTS(100, 0.65), 200);
+
+# Predict
+m_lin_reg_test = linear_reg(xs[1:2], Vector{Union{Missing, Float64}}(undef, 2));
+predictions = predict(m_lin_reg_test, chain_lin_reg)
+```
+results in
+```julia
+julia> predictions = predict(m_lin_reg_test, chain_lin_reg)
+Object of type Chains, with data of type 100×2×1 Array{Float64,3}
+
+Iterations        = 1:100
+Thinning interval = 1
+Chains            = 1
+Samples per chain = 100
+parameters        = y[1], y[2]
+
+2-element Array{ChainDataFrame,1}
+
+Summary Statistics
+  parameters    mean     std  naive_se     mcse       ess   r_hat
+  ──────────  ──────  ──────  ────────  ───────  ────────  ──────
+        y[1]  0.0078  0.0843    0.0084  missing   72.0724  1.0072
+        y[2]  0.1948  0.0950    0.0095  missing  131.6113  0.9969
+
+Quantiles
+  parameters     2.5%    25.0%   50.0%   75.0%   97.5%
+  ──────────  ───────  ───────  ──────  ──────  ──────
+        y[1]  -0.1581  -0.0567  0.0114  0.0550  0.1758
+        y[2]   0.0200   0.1198  0.2105  0.2667  0.3773
+
+```
+"""
```

Comment:
That's a nice and detailed documentation but could you reformat the docstring according to the [official guidelines](https://docs.julialang.org/en/latest/manual/documentation/)? E.g., use a single one-line sentence for describing the function in the imperative form and use a `# Examples` section with code put into a `julia-repl` or `jldoctest` code block? You could also use DocStringExtensions if you want to automate the display of the signature etc.

Reply:
Yeah of course; thanks for the comment! I didn't do `julia-repl` because the example takes up so much space already, but now I agree with you (+ the example in this PR above is a better one).